console/console-init/ui/src/AppRoutes.tsx

```diff
@@ -9,24 +9,26 @@ const getConnectionDetail = () => import("./Pages/ConnectionDetailPage");
 
 export const AppRoutes = () => (
   <SwitchWith404>
-    <Redirect path="/" to="/address-spaces" exact={true} />
+    <Redirect path="/" to="/address_spaces" exact={true} />
     <LazyRoute
-      path="/address-spaces"
+      path="/address_spaces"
       exact={true}
       getComponent={getAddressSpaceListPage}
     />
     <LazyRoute
-      path="/address-space/:id/:subList"
+      path="/address_space/name=:name&namespace=:namespace/:subList"
       exact={true}
       getComponent={getAddressSpaceDetail}
     />
     <LazyRoute
-      path="/address-space/:id/address/:id"
+      path="/address_space/name=:name&namespace=:namespace/address/name=:addressname&namespace=:addressnamespace"
```

Comment:
Same as above, plus, why the double namespace? Isn't it really the same you have higher in the root?

Wouldn't something like this work better? `/address-space/:namespace/:name/:addressname`

Reply:
we have to pass all the url param as filter to the query. So i have added name and namespace for both address and addressSpace in the route path